### PR TITLE
New schemas to handle Map

### PR DESCRIPTION
**Pull request by cfrancisco**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR introduces a whole new schema to handle the map components;

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

* **Other information**:
This PR is connected to dojot/dojot#701, dojot/dojot#676 and dojot/dojot#702.

### CODE REVIEW

File: src/js/views/utils/Maps.js [hunk 1]
```diff
@@ -312,174 +395,117 @@ class SmallPositionRenderer extends Component {
...
-                            <LayerBox
-                                key={lyr.id}
-                                toggleLayer={this.toggleLayer}
-                                config={lyr}
-                            />
-                        )) : null
-                }
+        return <div className="fix-map-bug">
+            <CustomMap toggleTracking={this.props.toggleTracking}  allowContextMenu={this.props.allowContextMenu} zoom={this.state.zoom} markersData={parsedEntries}/>
+            <div className="col s12 layer-box">
```
Comment (luisfelipefurlan) [inline src/js/views/utils/Maps.js]: These classes "col s12 layer-box" are causing misbehavior on the detail screen map.
image

After you change the zoom.

image

File: src/js/views/devices/DeviceMap.js [hunk 2]
```diff
@@ -24,32 +57,52 @@ class DeviceMap extends Component {
             mapquest: false,
```
Comment (luisfelipefurlan) [inline src/js/views/devices/DeviceMap.js]: Unnecessary property.

Comment (cfrancisco) [reply src/js/views/devices/DeviceMap.js]: Thanks! Removed.

File: src/js/views/utils/Maps.js (hunk 1 above)
Comment (cfrancisco) [reply src/js/views/utils/Maps.js]: You found the bug!! Thanks, I'll fix it.

**Pull request merged**